server/controllers/watchedController.js

```diff
@@ -70,120 +67,63 @@ exports.update = async function(req, res) {
+    } catch (err) {
+        res.status(RequestStatus.BAD_REQUEST).send(err);
+    }
 };
 
-var create_action = async function(user_id, watched_id) {
-    var action = new Action({
-        _user: user_id,
-        date: new Date(),
-        _action: watched_id,
-        action_type: WATCHED_ACTION_TYPE,
-    });
-    return action.save();
-}
-
-var delete_action = function(action_id) {
-    Action.remove({ _id: action_id}).exec();
-}
-
-var add_action_to_user_history = async function(user_id, action_id) {
-    User.findById(user_id, function (err, user) {
-        let user_history = user._history;
-        user_history.push(action_id);
-        return user.save();
-    });
-}
-
-var delete_action_from_user_history = async function(user_id, action_id) {
-    User.findById(user_id, function (err, user) {
-        let user_history = user._history;
-        let index = user_history.indexOf(action_id);
-        if (index > -1) {
-            user_history.splice(index, 1);
-        }
-        user.save();
-    });
-}
-
-var find_watched_obj = async function(watched_id) {
-    return Watched.findById(watched_id).exec();
-}
-
-var find_user_watched_list = async function(user_id) {
-    return Watched.find({_user: user_id}).exec();
-}
-
-
-getSeasonFromAPI = function(tv_id, season_number){
+var getSeasonFromAPI = function(tv_id, season_number){
   return new Promise(function(resolve, reject) {
-    var query = 'tvshow/' + tv_id + '/season/' + season_number
-    console.log("log")
+    var query = 'tvshow/' + tv_id + '/season/' + season_number;
     https.get("https://api.themoviedb.org/3/tv/"+ tv_id + "/season/"+ season_number +"?api_key=db00a671b1c278cd4fa362827dd02620",
     (resp) => {
       let data = '';
       resp.on('data', (chunk) => {
         data += chunk;
       });
       resp.on('end', () => {
-        console.log("saving season result to redis:"+  query)
+        console.log("saving season result to redis:"+  query);
         client.set(query, JSON.stringify(data));
         resolve(data)
       });
 
     }).on("error", (err) => {
-      console.log("Error: " + err.message);
       reject();
     });
   })
-}
-var inject_media_json = async function(watched_obj) {
-    let media_id = watched_obj._media;
-    let media_obj = await get_media_obj(media_id);
-    if (media_obj.__t == 'Episode' && media_obj._tmdb_tvshow_id){
-      var value = await getSeasonFromAPI(media_obj._tmdb_tvshow_id, media_obj.season_number).then((season)=>{
-        var watched_with_full_media = watched_obj;
-        watched_with_full_media._media = media_obj
+};
+
+var injectMediaJsonInWatched = async function(watched_obj) {
+    let mediaId = watched_obj._media;
+    let mediaObj = await DataStoreUtils.getMediaObjById(mediaId);
+    if (mediaObj.__t == 'Episode' && mediaObj._tmdb_tvshow_id){
+      var value = await getSeasonFromAPI(media_obj._tmdb_tvshow_id, media_obj.season_number).then((season) => {
+        var watched_with_full_media = watchedObj;
```

Comment:
A função recebe como parametro watched_obj, mas aqui é usado watchedObj causando erro.

Reply:
Thank you. Resolved